components/ResizeableWindow.jsx

```diff
@@ -142,12 +142,18 @@ class ResizeableWindow extends React.Component {
             "resizeable-window-body-scrollable": this.props.scrollable,
             "resizeable-window-body-nonscrollable": !this.props.scrollable
         });
-        const style = {display: this.props.visible ? 'initial' : 'none'};
+        const style = { display: this.props.visible ? 'initial' : 'none' };
         const maximized = this.state.geometry.maximized ? true : false;
         const minimized = this.state.geometry.minimized ? true : false;
         const zIndex = 10 + this.props.windowStacking.findIndex(item => item === this.id);
-        const dockBottom  = this.props.dockable === "bottom";
-        const dockIconSuffix = dockBottom ? "_bottom" : "";
+        let docked = this.state.geometry.docked;
+        const dockSide = this.props.dockable === true ? "left" : this.props.dockable;
+        const dockIconSuffix = (dockSide === "bottom" || dockSide === "top") ? "_bottom" : "";
+        let dockIcon = docked ? 'undock' : 'dock';
+        if (dockSide === "right" || dockSide === "top") {
+            dockIcon = docked? 'dock' : 'undock';
+        }
```

Comment:
I think it would be better to provide dedicated icons for dock/undock top/right.

Reply:
Yes, I did that in [this commit](https://github.com/qgis/qwc2/pull/226/commits/433f65077b953f7f2608da9620e3ad610f7acfb9)